### ISSUE TITLE
Keybase chat hanging on all devices

### ISSUE THREAD

**Issue by zonotope** [keybase/client#24208]
Hi. I have 2 Linux laptops and an Android phone each with the Keybase client installed. This morning (EST) all of them stopped responding. I've gotten notifications that people have sent me chat messages, but my devices haven't been able to decrypt for a few hours. I've uninstalled and reinstalled keybase on my phone, but now I just get a spinner and then a screen with no text and an image of a woman in what looks like a yoga/meditation pose, and the only call to action is a button that says "close".

I spoke to other members of my team, and they report no issues with their Keybase clients. 

Is there a problem with the keybase servers? Is it my network connection (I was just on a google hangouts call and accessing the Internet through a browser seems fine and fast)? 

As it stands now, I cannot access keybase or use it to communicate with my team, and haven't been able to all morning. How do I fix this?

**Comment by chindraba-work** [comment]
First step, don't uninstall anything else! Period! As it stands now your phone is no longer a valid device for your account because it will have to be provisioned again, with a new name. Hopefully you have a paper key you can do that with. If no paper key is available you will need one of the Linux computer to authorize the phone.

**Comment by zonotope** [comment]
Ok, thanks. It might be difficult for me to find the paper key, but like I said, the linux computers are also not responding when I try to access them. I just get spinners. Even when I try to log in on the phone after uninstalling it, there's no place to enter the paper key. I just get a cartoon of a woman in a meditation pose with a close button. It seems weird for all of my devices to stop working at the same time.

I also tried to chat with your support but, alas, It seems I can only do that through the Keybase app that doesn't work.

**Comment by chindraba-work** [comment]
Ok, authorization out of the way, have you tried using the command line for Keybase on the laptop? Much better error messages in the command line than in the GUI. Also, are the laptops up-to-date with the Keybase version? Current version is 5.5.1. Using your system's tools kill all keybase processes, if any. Then try `run_keybase` on the command line.

**Comment by zonotope** [comment]
I am on the latest version:
```
% >> keybase --version                             
keybase version 5.5.1
```
`systemctl --user stop keybase` was hanging, but I manually killed all the keybase related processes except for the one that systemd kept restarting, but then `systemctl --user stop keybase` worked to stop that remaining process:
```
% >> ps aux | grep keybase                         
zonotope    6282  0.0  0.0   6296  2356 pts/1    S+   12:48   0:00 grep --color=auto --exclude-dir=.git keybase
```
I get "command not found" for `run_keybase` however. Is there another command I should use?

**Comment by zonotope** [comment]
I'm on Arch Linux by the way, and I installed keybase through the `keybase` and `keybase-gui` arch packages.

**Comment by chindraba-work** [comment]
Ok, I've read that Arch did it somewhat differently. It appears that the command is `keybase-bin`

**Comment by chindraba-work** [comment]
It looks like `keybase-gui` starts the GUI, and `keybase` handles the CLI. Not so sure about the daemon, though if systemd kept restarting it there has to be a `.desktop` file for it, and that should help figure out the command(s).

**Comment by chindraba-work** [comment]
If you can get the CLI running, at the least, try `keybase ping` and see if you get a response from the server.

**Comment by zonotope** [comment]
yeah, it's `keybase-gui`. here's the contents of `/usr/share/applications/keybase.desktop`:
```[Desktop Entry]
Name=Keybase
Exec=keybase-gui %u
Icon=keybase
Terminal=false
Type=Application
Categories=Network;
StartupNotify=false
MimeType=x-scheme-handler/web+stellar;x-scheme-handler/keybase;application/x-saltpack;
```

and here's the contents of `~/.config/autostart/keybase_autostart.desktop`:
```
# This file is generated the first time Keybase starts, along with a sentinel
# file at ~/.config/keybase/autostart_created. As long as the sentinel exists,
# this file won't be automatically recreated, so you can edit it or delete it
# as you like. Manually changing autostart settings in the GUI can stomp on
# your edits here though.

[Desktop Entry]
Name=Keybase
Comment=Keybase Filesystem Service and GUI
Type=Application
Exec=env KEYBASE_AUTOSTART=1 keybase-gui
```
I'm not sure how to get the cli running independent of the gui, but I tried `keybase ping` with everything killed anyway and I got this, which has a bit more info at least:

```
% >> keybase ping                                                                                                                 
▶ INFO Starting keybase.service.
▶ ERROR API network error: Post "https://api-0.core.keybaseapi.com/_/api/1.0/ping.json": dial tcp 54.174.212.80:443: i/o timeout

took 30s
-```

**Comment by chindraba-work** [comment]
Interesting, and seems to be the base of the problem.

Somehow your keybase client is trying to connect to a down server. I get this:

```
chindraba@Meska:~> keybase ping
▶ INFO API Server at https://api-0.core.keybaseapi.com is up
chindraba@Meska:~> nslookup api-0.core.keybaseapi.com
Server:         172.21.13.69
Address:        172.21.13.69#53

Non-authoritative answer:
Name:   api-0.core.keybaseapi.com
Address: 52.4.235.66
Name:   api-0.core.keybaseapi.com
Address: 52.45.72.22

chindraba@Meska:~> ping 54.174.212.80
PING 54.174.212.80 (54.174.212.80) 56(84) bytes of data.
^C
--- 54.174.212.80 ping statistics ---
5 packets transmitted, 0 received, 100% packet loss, time 4097ms

chindraba@Meska:~> nslookup 54.174.212.80
80.212.174.54.in-addr.arpa      name = ec2-54-174-212-80.compute-1.amazonaws.com.

Authoritative answers can be found from:
```

I have no idea how to re-home the client. It's probably done based on an internal heuristic for finding the closest AWS instance. I also have no idea how to help, and can only suggest having patience.

This could be a sign of the times with Keybase now being a Zoom product. Zoom might not be paying the price to keep the instances alive.

**Comment by zonotope** [comment]
:disappointed: 

Thanks for the help, but that's a real bummer. I use keybase for work mostly and, of course, everyone is remote right now, so I'm currently disconnected from my team. Any suggestion on how I can get any of my devices to decide to connect to a live server soon would be greatly appreciated.

**Comment by chindraba-work** [comment]
Without knowing how it chooses anyway the only work-around I can think of is to try a vpn to a different geo in hopes that it will load-balance to a different server, or place an entry in the `/etc/host` file (or any other resolver you have access to) to try forcing `api-0.core.keybaseapi.com` to resolve to `52.45.72.22`, which is at the moment working for me.

**Comment by chindraba-work** [comment]
As an additional thought, this is a good reminder that Keybase has no guaranteed future and setting up a secondary, or fallback, channel or at least an emergency alternate channel, would be a good idea. Even if it's only a Discord server where you can say, "Hey, my Keybase is down again."

**Comment by zonotope** [comment]
Thanks a lot for your help. The hosts workaround worked for now. I really hope keybase doesn't go away because it's such a great product. I can also see how it would be beneficial if Zoom integrated it into their core products. They could _own_ secure communications if they played their cards right.

**Comment by chindraba-work** [comment]
I guess that means you're not up to date on Zoom's stance: 
https://twitter.com/NicoAGrant/status/1268020841054269440
